pilot/cmd/pilot-agent/main.go

```diff
@@ -290,6 +291,9 @@ var (
 			stop := make(chan struct{})
 			cmd.WaitSignal(stop)
 			<-stop
+			http.Post(fmt.Sprintf("http://127.0.0.1:%d/healthcheck/fail", proxyAdminPort), "", nil)
```

Comment:
Did Istio Proxy enable active health checks? IIRC, @rshriram mentioned that Istio Proxy has not enabled active health checks. The active `healthcheck/fail `would only work if active health checking is enabled.

Reply:
That's correct, Istio is not currently using active health checks.

@hzxuzhonghu have you tested this?

Reply:
we generate healthcheck filters for kubernetes probes so that they are not traced. So this is what @hzxuzhonghu is going by.

Reply:
@rshriram ok, I think I'm a little out of the loop with what's been going on in this area ... maybe you can set me straight...

I'm assuming that the `ManagementPorts` are exposed by the application container and receive probes from k8s ... which are then redirected to Envoy and get caught by these filters? It appears that the draining logic won't occur until after the container receives a SIGTERM. If that's correct, what exactly does this PR do? It would seem that draining after SIGTERM would be too late to have any impact, no?

Reply:
May be I am missing some thing here, but If I read this correctly, this is just enabling health checks for ingress listeners. I think the draining would work only if the health check is enabled for upstream clusters and upstream hosts would immediately send` x-envoy-immediate-health-check-fail` so that the host is removed from load balancer and outgoing calls would drain properly on the host but it wont accept incoming requests. Not sure it the same happens here if we just enable it for ingress listeners.

Are you attempting to do some other thing here?